### ISSUE TITLE
Remove requirement to have 2 names in CLA signature

### ISSUE THREAD

**Issue by kborchers** [jquery/jquery-license#54]
I have tried making this argument before and I think this PR sums up my argument nicely https://github.com/jquery/esprima/pull/1331

**Comment by scottgonzalez** [comment]
There is no requirement for 2 names, there's only a requirement that single name signatures are moderated.

With that being said, feel free to make whatever changes you want.

**Comment by michaelficarra** [comment]
> There is no requirement for 2 names ...

[Yeah there is.](https://github.com/jquery/jquery-license/blob/cdd4730760de73619c386927c54462e2c0fd41e5/lib/validate.js#L19)

**Comment by scottgonzalez** [comment]
[No, there isn't.](https://github.com/jquery/jquery-license/blob/88556c783aa429787b9eb4b65467639808893d3d/lib/signatures.js#L61-L64)

**Comment by kborchers** [comment]
To Michael's point though, it is not welcoming and I would agree even a possible CoC issue to question what someone says is their name. I think our best option here is to clearly state that we want a legal name on the CLA signature and if they state that is their legal name, who are we to question that? I will also run this by our legal folks to get their opinion but that is where I stand right now. Any thoughts from @jquery/conduct or @jquery/directors? /cc @stevenayr

**Comment by michaelficarra** [comment]
@scottgonzalez A potential contributor was rejected with the failure "Suspicious name: fishbar". How would you explain that validation failure, considering that is the only place that text appears in this repo? The requirement *is* being enforced, and that is the code that is doing it. I don't know what your link to its call site was supposed to show.

**Comment by mikesherov** [comment]
Right, @michaelficarra, this is more of a language thing than I workflow thing. If the text said "Thanks for contributing. Please wait while we manually approve your CLA signature", that essentially solves it. "suspicious name: fishbar" is unwelcoming.

**Comment by scottgonzalez** [comment]
> I think I see now. You're not saying the validation does not happen, you're saying there is a way for CLA signatures to be manually accepted even with validation failures.

Correct. The official list of CLA signatures has a field that allows moderators to mark a name as valid even if it would fail the standard check. If the confirmation field is filled in by a moderator (the code I linked to), then the standard validation won't run against that name (the code you linked to).

**Comment by scottgonzalez** [comment]
> A potential contributor was rejected with the failure "Suspicious name: fishbar"

That's very strong wording. Neither the contributor nor the contribution has been rejected. An automated check failed on the first run and the issue has already been resolved. It's worth nothing that the failure would have occurred on the first run anyway since the user signed the CLA with a different pseudonym than the one used in their git config.

**Comment by mikesherov** [comment]
The failure messaging is the question here. "Suspicious name" reads as unwelcoming to me.

**Comment by gabrielschulhof** [comment]
http://www.bbc.com/news/blogs-trending-31699618

**Comment by jzaefferer** [comment]
PS: Updating the logic in `lib/validate.js` is trivial, both to remove the check and to change the message.

**Comment by jzaefferer** [comment]
@kborchers what's the verdict?

**Comment by kborchers** [comment]
After a chat with legal here is where we stand. By removing the need for 2 names we do not significantly increase the likelihood that someone will provide fake information. If their intent is to provide something other than their legal name, they can do that with 2 names as easy as 1. With that in mind, legal recommends we make an update to the CLA signature form to request their legal name just to be clear that is what we want and to remove the check for 2 names. Legal also requested that if we could request some other form of non-sensitive but personally identifiable information that would be helpful (like an IP address or something like that though I don't know if that's even possible within our current system). I think for the sake of inclusiveness we should not require 2 names or even question someone who provides a single name as it is not our place to question someone's legal name as long as that is what we are requesting and by signing they are stating that is what they are providing. If there are questions or concerns for me or legal please feel free to comment further before we make this change and close this issue but we should try to make this change ASAP without any objection in the next day or two.

**Comment by gnarf** [comment]
Can I add for consideration that "legal name" could also be a barrier to
submissions for transitioning individuals?

What is it we actually need here?

**Comment by scottgonzalez** [comment]
> By removing the need for 2 names we do not significantly increase the likelihood that someone will provide fake information

This isn't just about fake information. And it's certainly not about contributors intentionally providing misleading information. Out of curiosity, what is this assumption based on? The actual signatures that we've received, actual knowledge of prior similar online forms, or purely a guess? Because based on my personal experience with the existing data, removing the check does significantly increase the chances that we end up with bad author data in our repositories.

**Comment by michaelficarra** [comment]
@scottgonzalez See the sentence immediately following the one you quoted.

> If their intent is to provide something other than their legal name, they can do that with 2 names as easy as 1.

**Comment by scottgonzalez** [comment]
@michaelficarra Yes, and that's fine. We don't actually care about legal names.

**Comment by gnarf** [comment]
It's about the person who's `.gitconfig` has `author.name="gnarf"` and is
just like - ah, I'll just sign this form as "gnarf" since it must match!

**Comment by kborchers** [comment]
We need something to be able to track someone down in the unlikely case that someone submits code that is not owned by them and we are being pursued legally by the rightful owner to understand what happened and determine next steps. I honestly don't care what we collect as long as it gets us to the person that submitted it. Without the ideal situation of a physical signature on a piece of paper we have to take people at their word at some point but we need to try to find a way to actually be able to reach those people if necessary. I'm not sure if there is a perfect answer but I would like to try to find the right balance of inclusiveness and legal protection of our project's code bases.

**Comment by gnarf** [comment]
I'll bring this up from @mikesherov again because I think this might be the solution.

> If the text said "Thanks for contributing. Please wait while we manually approve your CLA signature", that essentially solves it. "suspicious name: fishbar" is unwelcoming.

**Comment by kborchers** [comment]
Maybe changing the messaging is the right way. I was just trying to help our project maintainers avoid the need to constantly verify names but if that is what we deem best, and our conduct team believes this is the solution most inline with our CoC, I am fine with that.

**Comment by mikesherov** [comment]
Message change ftw the win. There is no more information in "suspicious name: fishbar" and "Thanks for contributing. Please wait while we manually approve your CLA signature: fishbar".

**Comment by gnarf** [comment]
So the validation check is not the problem them @kborchers?  Just the language?

People are highly unlikely to have a 'single name', usually when they do it's because it's their username on their computer and they just sign with it because it was what the git commit already had.

Basically - we want to verify that email / there is a person named "fishbar" that will answer it, when this case does come up, in order to stop the lazy people (which we had more than a few of) from giving us bogus information.  The trade off.  Do we ask single named people to wait a bit while we check a box on a sheet (trivial to do when accepting a PR), or do we open ourselves up to lazy people just using their computer username.

**Comment by scottgonzalez** [comment]
As a project maintainer, I would much rather verify a name than let the authors list become a mess.

**Comment by scottgonzalez** [comment]
> People are highly unlikely to have a 'single name'

In many cultures that's true, but certainly not all. Even within a culture that generally uses multiple names, any individual may choose to identify themselves by only a single name. This is why I implemented the manual acceptance process. It can certainly be improved upon with some smarter logic based on which characters are being used, but so far this hasn't really been an issue.

**Comment by kborchers** [comment]
@gnarf it seems that way. This was transferred from the Esprima issue mentioned above and maybe we went down the technical path of a solution when a language change was good enough. Does the suggested language from @mikesherov work for you @michaelficarra @gnarf @scottgonzalez ?

**Comment by scottgonzalez** [comment]
> Does the suggested language from @mikesherov work for you @michaelficarra @gnarf @scottgonzalez ?

yes

**Comment by gnarf** [comment]
So let's switch up the messaging.  Take what @mikesherov said and link to a section about this exception in our documentation.  Something like

> Thanks for contributing. Please wait while we [manually approve](#) your CLA signature: fishbar. 

and then the "manually approve" is a heading on the CLA page:

> If you use a single name as your real name, we need to manually approve it before accepting the commit. Many contributors in the past have created commits using their computer usernames as the real name.  In order to keep a consistent authors list, we request that you sign your commits with your real name instead of your username.

@michaelficarra @kborchers @scottgonzalez - Can we ask people with single names what they think of this wording?

**Comment by kborchers** [comment]
Assuming we agree, something that may be helpful for maintainers and contributors is defining an appropriate verification process. Asking someone "Is this your name?" or something like that could be perceived as hostile. Sharing methods you have used as maintainers @scottgonzalez @mikesherov could be very helpful to others. Also, I like @gnarf's suggestion to try to find folks with single names and get their opinion of the wording.

**Comment by gnarf** [comment]
cc @jquery/conduct - Please take a look at this issue and the proposed wording?

**Comment by chaegribben** [comment]
So what if Prince started keeping his lyrics under version control....I mean The Artist Formally Known as Prince........I mean that symbol guy.....nevermind.

**Comment by gnarf** [comment]
> So what if Prince started keeping his lyrics under version control....I mean The Artist Formally Known as Prince........I mean that symbol guy.....nevermind.

We'd manually approve the name when we accepted the commit.  Please, lets try to keep the discussion here on topic. :smile_cat:

**Comment by michaelficarra** [comment]
@chaegribben Please don't make light of this issue. Names are very meaningful to some people, especially in certain cultures.

**Comment by chaegribben** [comment]
It is on topic and I wasn't really making light - excuse my attempt at making a point through humour - this is likely to be a difficult problem to solve whilst taking all possible edge cases (as Prince would be) into account. So if it's that important to a user they may still have to accept a middle-ground. Manual approval will still be necessary in a lot of cases if we're trying to cater to all.

**Comment by scottgonzalez** [comment]
Assuming that Prince wasn't famous and therefore we didn't know who he was...

If the contribution came while his name was Jamie Starr, Christopher Alexander Nevermind, The Purple One, or Joey Coco, the name would have passed the automated check. Note that The Purple One is a perfect example of a name that would pass the automated check but would raise suspicion during a manual review.

If the contribution came while he was Prince, he would have signed with Prince and we would've asked for confirmation that Prince is the name he goes by since it would fail the automated check.

If the contribution came while he was using Love Symbol #2, I honestly don't know how he would've signed his name, since he can't write his name in unicode. This is actually a problem for many people in the world, and we just have to accept whatever written form they choose to use to represent their name in unicode.

**Comment by jzaefferer** [comment]
#56 proposes a new error message that replaces the somewhat hostile language with a hint at the manual verification. Please take a look.

**Comment by ashleygwilliams** [comment]
how about a checkbox labelled "the written representation of my name does not contain a space"

also include a clientside warning when no space is entered in the name field and box is not checked.

warning could read: "you did not enter a name containing a space. please include your full legal name, and check the box if it does not include a space."

**Comment by gnarf** [comment]
I do like Ashley's proposal, but I'm still worried the 'lazy' will use this
exception to submit usernames.  I just watched a friend get lost trying to
sign with a @noreply email as well.  Anything to not have to ammend the
commit with new author, and was frustrated when I said they'd have to.

I'm also not super sure how much control we have over the google forms
submission (we might already have a custom form solution, but I can't
remember and am on mobile right now)

how about a checkbox labelled "the written representation of my name does
not contain a space"

also include a clientside warning when no space is entered in the name
field and box is not checked.

warning could read: "you did not enter a name containing a space. please
include your full legal name, and check the box if it does not include a
space."

—
Reply to this email directly or view it on GitHub
<https://github.com/jquery/jquery-license/issues/54#issuecomment-149039367>.

**Comment by scottgonzalez** [comment]
The form is completely custom and already contains some validation, so the additional checkbox and client-side validation logic isn't an issue.

@ashleygwilliams and I went through a few different options before she posted that, so I'm +1 on her suggestion.

**Comment by gnarf** [comment]
Ok. Just to verify, would that proposal remove the need to have maintainers
manually approve these names?  I'm pretty sure that within a few months
someone is going to use that 'exception' to sign with their username.

**Comment by scottgonzalez** [comment]
@ashleygwilliams' proposal was that the checkbox would avoid the need for the manual verification, but we would occasionally do a manual review of all the names that used the checkbox and see if people are abusing it. If they are, we can revisit.

**Comment by gnarf** [comment]
+1

**Comment by mikesherov** [comment]
:+1:

**Comment by gnarf** [comment]
Automation fail... Reopened

**Comment by jzaefferer** [comment]
I don't understand why this got closed, the commit message only had a "Ref #54".

**Comment by gnarf** [comment]
I think it was because the body of the PR itself had 'Fixes'

**Comment by StommePoes** [comment]
My cheap opinion of the situation is, using (lack of) spaces as a placeholder to a separate problem is kinda silly, and starts going along the CAPTCHA road.* You want people to use a unique, preferably "real" name *so you can reliably contact them in the future if necessary*, so just say that. Stating you have a problem and want their help is going to be more "friendly" (for whatever measure of "friendliness" you're going for) than telling people there's something wrong with their names. 

If you empirically see that you're getting a lot of obviously-computer-names submitted, you can still add a front-end validation that, when it notices a one-word name, does nothing more than a popup/alert restating how important it is to you that they're not computer names/ internet handles which still doesn't prevent users from submitting. It's the best you can do: plead to the lazy, but you cannot stop the deliberate.

In any case, don't tell people their names lack spaces/have weird characters/have words that match a naughty-word dictionary. Even if you want to still have manual checks in the back. Do you have to tell the user you're checking? If nobody tells Robin Kills The Enemy her account name was eyeballed by someone before approval, does she care so long as she gets approved in a timely fashion (and does she have easy and open recourse should it be rejected?) 

I have two passports. They each have different names, and they're both legal and current (and one of them is horribly long, ew) . Do you want to go down that rabbit hole?


[imagine a footnotey asterisk here]CAPTCHA road: You want to prevent  "solving" the problem with something mostly unrelated. I think CAPTCHAs are an excellent example of a poor solution to some problem it doesn't really solve very well (when we want to know if a user is *legitimate*, we do not care if the user is *sweating pooping human*, whereas CAPTCHA straight-up claims to be used to prove humanness rather than legitimacy. Thus, in most applications of CAPTCHA, legitimate users such as (sometimes) scripts and people using screen readers are blocked, while sweating-pooping Mechanical Turks and sweating-pooping human spammers are let in: quite the opposite of what we want).
